pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/ApexTreeBuilder.kt

```diff
@@ -0,0 +1,134 @@
+/*
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.apex.ast
+
+import net.sourceforge.pmd.annotation.InternalApi
+import net.sourceforge.pmd.lang.apex.ApexParserOptions
+import net.sourceforge.pmd.lang.ast.ParseException
+import net.sourceforge.pmd.lang.ast.SourceCodePositioner
+
+import com.google.summit.ast.CompilationUnit
+import com.google.summit.ast.Identifier
+import com.google.summit.ast.Node
+import com.google.summit.ast.TypeRef
+import com.google.summit.ast.declaration.ClassDeclaration
+import com.google.summit.ast.declaration.EnumDeclaration
+import com.google.summit.ast.declaration.InterfaceDeclaration
+import com.google.summit.ast.declaration.TriggerDeclaration
+import com.google.summit.ast.declaration.TypeDeclaration
+import com.google.summit.ast.modifier.KeywordModifier
+import com.google.summit.ast.modifier.Modifier
+
+@Deprecated("internal")
+@InternalApi
+@Suppress("DEPRECATION")
+class ApexTreeBuilder(val sourceCode: String, val parserOptions: ApexParserOptions) {
```

Comment:
I'm concerned that your converting existing classes to kotlin will make the merge conflicts with pmd 7 much harder to handle. The java version of this class has been changed in pmd 7 and will conflict. I think it's fine to use kotlin for new classes but I see no solid reason to convert existing classes to kotlin when they already work - at least while we have three parallel development branches.

Reply:
This class serves the same functional purpose but is essentially a wholesale rewrite. (The input is obviously different, but the general structure changed to handle the many:many relation between the two ASTs.)

Any concerns about specific common pieces that we might want to factor out and share? I can rework to preserve those in ApexTreeBuilder and move the Summit-specific bits to a new class.